recorder.go

```diff
@@ -377,29 +378,18 @@ func convertToKeyValue(k string, value interface{}) *cpb.KeyValue {
 	return &kv
 }
 
-// TODO: Decide what we want to do about duplicate keys
-func translatePayload(pl interface{}, kvs []*cpb.KeyValue) []*cpb.KeyValue {
-	switch p := pl.(type) {
-	case map[string]interface{}:
-		for k, v := range p {
-			kvs = append(kvs, convertToKeyValue(k, v))
-		}
-	case []interface{}:
-		for _, v := range p {
-			kvs = append(kvs, convertToKeyValue(payloadKey, v))
-		}
-	case nil:
-		return nil
-	default:
-		kvs = append(kvs, convertToKeyValue(payloadKey, p))
-	}
-	return kvs
-}
-
 func translateEventAndPayload(e string, pl interface{}) []*cpb.KeyValue {
 	var kvs []*cpb.KeyValue
 	kvs = append(kvs, &cpb.KeyValue{Key: messageKey, Value: &cpb.KeyValue_StringValue{e}})
-	return translatePayload(pl, kvs)
+	if pl == nil {
+		return kvs
+	}
+	jpl, err := json.Marshal(pl)
+	glog.Infof("the log is %v", string(jpl))
+	if err != nil {
+		return append(kvs, &cpb.KeyValue{Key: payloadKey, Value: &cpb.KeyValue_StringValue{fmt.Sprintf("%v", err)}})
```

Comment:
@djspoons, @jmacd and I were thinking that since this error comes from marshaling user sent data, it might make sense for them to be aware of that. But I don't think we are too tied to the idea if you disagree.